webapp/routes/api.js

```diff
@@ -63,9 +63,14 @@ router.post('/users/authenticate', function (req, res, next) {
         if (!user) {
             return next(new Error("User not found"));
         }
+
+        if(user.password != req.body.password) {
+            return next(new Error("Incorrect Password"));
+        }
+
         user.password = null;
         res.json(user);
```

Comment:
I don't think so. The behavior is ok in the api but wasn't in socket. I will add the resetting of the password just in case though.